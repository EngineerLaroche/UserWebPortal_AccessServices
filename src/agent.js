import superagentPromise from 'superagent-promise';
import _superagent from 'superagent';

const superagent = superagentPromise(_superagent, global.Promise);
const USER_API = 'http://localhost:4000';

//const encode = encodeURIComponent;
const responseBody = res => res.body;

let token = null;
const tokenPlugin = req => {
  if (token) {
    req.set('authorization', `Token ${token}`);
  }
}

const requests = {
  del: url =>
    superagent.del(`${USER_API}${url}`).use(tokenPlugin).then(responseBody),
  get: url =>
    superagent.get(`${USER_API}${url}`).use(tokenPlugin).then(responseBody),
  put: (url, body) =>
    superagent.put(`${USER_API}${url}`, body).use(tokenPlugin).then(responseBody),
  post: (url, body) =>
    superagent.post(`${USER_API}${url}`, body).use(tokenPlugin).set('Content-Type', 'application/json').then(responseBody)
};

export const Auth = {
  current: () =>
    requests.get('/user'),
  login: (email, password) =>
    requests.post('/user/login', { user: { email, password } }),
  register: (firstname, lastname, email, password, role) =>
    requests.post('/user/create', { user: { firstname, lastname, email, password, role } }),
  logout: () =>
    requests.post('/user/logout', {}),
  save: user =>
    requests.put('/user/update', { user }),
  deactivate: (email) =>
    requests.put('/user/deactivate', { email }),
  reactivate: (email) =>
    requests.put('/user/reactivate', { email })
};

const Referent = {
  register: (firstname, lastname, title, workphone, cellphone, fax, email, preferences) =>
    requests.post('/referent/create', { referent: { firstname, lastname, title, workphone, cellphone, fax, email, preferences } }),
  associate: (organismeId, email) =>
    requests.post('/referent/organismereferent', { association: { organismeId, email } }),
  all: () =>
    requests.post('/referents', {}),
  save: referent =>
    requests.put('/referent/update', { referent }),
  clear: (email) =>
    requests.del('/referent/clearpreferences', { association: email }),
  delete: id =>
    requests.del('/referent/delete', { id }),
  preference: (preferenceId, email) =>
    requests.post('/referent/preferencereferent', { association: { preferenceId, email } })
};

const OrganismeReferent = {
  register: (name, email, address, phone, fax, website, state, organisme) =>
    requests.post('/organismereferent/create', { organismeReferent: { name, email, address, phone, fax, website, state, organisme } }),
  all: () =>
    requests.post('/organismereferents', {}),
  save: organismeReferent =>
    requests.put('/organismereferent/update', { organismeReferent }),
  delete: id =>
    requests.del('/organismereferent/delete', { id })
};

const Organisme = {
  register: (name, email, address, phone, fax) =>
    requests.post('/organisme/create', { organisme: { name, email, address, phone, fax } }),
  all: () =>
    requests.post('/organismes', {}),
  save: organisme =>
    requests.put('/organisme/update', { organisme }),
  delete: id =>
    requests.del('/organisme/delete', { id })
};

const PointService = {
  register: (name, email, phone, fax, nocivique, street, city, province, postalcode) =>
    requests.post('/pointservice/create', { pointService: { name, email, phone, fax, nocivique, street, city, province, postalcode} }),
  all: () =>
    requests.post('/pointservices', {}),
  save: pointService =>
    requests.put('/pointservice/update', { pointService }),
  delete: id =>
    requests.del('/pointservice/delete', { id })
};

const Service = {
  register: (name, description, tarifParent, tarifCISSS, state, stateSubvention, datePrice) =>
    requests.post('/service/create', { service: { name, description, tarifParent, tarifCISSS, state, stateSubvention, datePrice} }),
  all: () =>
    requests.post('/services', {}),
  save: service =>
    requests.put('/service/update', { service }),
  delete: id =>
    requests.del('/service/delete', { id })
};

const DemandeService = {
  register: (serviceType, noDossierFamille, date, frequence, assumeFrais, motifs, transport, parent, secondParent, enfants, dispoFirstParent, dispoSecondParent) =>
    requests.post('/demandeservice/create', { demandeservice: { serviceType, noDossierFamille, date, frequence, assumeFrais, motifs, transport, parent, secondParent, enfants, dispoFirstParent, dispoSecondParent} }),
  all: () =>
    requests.post('/demandeservices', {}),
  save: (id, date, frequence, assumeFrais, motifs, transport, parent, secondParent, enfants, dispoFirstParent, dispoSecondParent) =>
    requests.put('/demandeservice/update', { demandeservice: { id, date, frequence, assumeFrais, motifs, transport, parent, secondParent, enfants, dispoFirstParent, dispoSecondParent} }),
  delete: id =>
    requests.del('/demandeservice/delete', { id }),
  addMotif: (motif) =>
    requests.post('/motifs/add', {newMotif: motif})
};

const Users = {
  all: () =>
    requests.get('/users')
};

export default {
  Auth,
  Users,
  Referent,
  OrganismeReferent,
  Organisme,
  PointService,
  Service,
  DemandeService,
  setToken: _token => { token = _token; }
};
