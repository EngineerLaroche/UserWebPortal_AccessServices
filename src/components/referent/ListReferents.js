import { Link } from "react-router-dom";
import Request from 'superagent';
import React from 'react';
import { connect } from 'react-redux';
import { GETREFERENTS } from '../../constants/actionTypes';

const mapStateToProps = state => ({
    referents: '',
    search: ''
});

const mapDispatchToProps = dispatch => ({
    onLoad: () =>
        dispatch({ type: GETREFERENTS })
});

export class ListReferents extends React.Component {
    constructor() {
        super();
        this.state = {
            search: '',
            referents: []
        }
    }

    searchReferents(query) {
        var url = "http://localhost:4000/referent/search/" + query;
        Request.get(url).then((response) => {
            console.log(response.body);
            var data = response.body.returnedReferents;
            this.setState({ 'referents': data });
        });
    }

    componentWillMount() {
        var url = "http://localhost:4000/referents"
        Request.get(url).then((response) => {
            console.log(response.body);

            var data = response.body.referents;

            this.setState({ 'referents': data });
        });
    }

    handleChange(value, param) {
        this.setState({ [param]: value });
    }

    render() {
        var data = this.state.referents;
        var search = this.state.search;

        return (

            <div className="container page">

                <hr />
                <h1 className="text-xs-center">Liste des Référents</h1>
                <hr />
                <br />
                <br />

                <div className="row">
                    <div className="col-md-8 offset-md-2 col-xs-12">
                        <div className="text-xs-center">
                            
                            <label>Search: </label>
                            <br/>
                            <fieldset>
                                <fieldset className="form-group">
                                    <input
                                        type="text"
                                        value={this.search}
                                        onChange={(e) => { this.handleChange(e.target.value, 'search') }} />
                                    <button onClick={() => 
                                        { this.searchReferents(search) }}>Go</button>
                                </fieldset>
                            </fieldset>
                            </div>
                            <br/>
                            <br/>
                            <div className="text-xs-center">
                            {(data != null) && Object.values(data).map((d) =>
                                <li key={d.Email}>
                                    <Link to={"/referent/" + d.Email} className="nav-link">
                                        {d.Firstname} {d.Lastname}
                                    </Link>
                                    <hr/>
                                </li>
                            )}
                            {(data != null && data.length === 0) &&
                                <p>Aucun référent trouvé</p>
                            }
                        </div>
                    </div>
                </div>
            </div>
        );
    }
} export default connect(mapStateToProps, mapDispatchToProps)(ListReferents);