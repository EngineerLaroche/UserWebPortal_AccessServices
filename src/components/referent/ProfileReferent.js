import React from 'react';
import { Link } from 'react-router-dom';
import Request from 'superagent';
import { connect } from 'react-redux';

const EditProfileReferentSettings = props => {
  if (props.isReferent) {
    return (
      <Link to={`/editreferent/${props.currentReferent.email}`} className="btn btn-primary">
        <i className="ion-gear-a"></i> 
        Modifier Référent
      </Link>
    );
  }
};

const mapStateToProps = state => ({
  currentReferent: state.common.currentReferent,
  referent: state.referent
});

const mapDispatchToProps = dispatch => ({  });

export class Referent extends React.Component {

  constructor(){
    super();
    this.state = {
      currentReferent: ''
    }
  }

  componentWillMount() {
    var url = "http://localhost:4000/referent/" + this.props.match.params.email;
    Request.get(url).then((response) => {
      console.log(response.body);
      var data = response.body.referent;
      this.setState({'referent':data});
      this.setState({'currentReferent':data});
    });
  }

  render() {
    const referent = this.state.currentReferent;
    if (!referent) {
      return null;
    }

    const name = referent.firstname + " " + referent.lastname;
    
    return (
      <div className="referent-page">
        <div className="referent-info">
          <br/>
          <br/>
          <h1 className="text-xs-center">Profil du référent</h1>
          <br/>
          <br/>
          <div className="container">
            <div className="row">
              <div className="text-xs-left">

                <h4>Nom: {name}</h4>
                <h4>Email: {referent.email}</h4>
                <h4>Titre: {referent.title}</h4>
                <br/>
                <h4>Telephone travail: {referent.workphone}</h4>
                <h4>Telephone cellulaire: {referent.cellphone}</h4>
                <h4>Fax: {referent.fax}</h4>
                <h4>Preference: {referent.preference}</h4>
                <EditProfileReferentSettings isReferent={true} 
                                              currentReferent={referent} 
                                              onClickDeactivate={ () => {this.props.onClickDeactivate(referent.email)}} 
                                              onClickReactivate={() => {this.props.onClickReactivate(referent.email)}}/>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Referent);
//export { Referent, mapStateToProps };
