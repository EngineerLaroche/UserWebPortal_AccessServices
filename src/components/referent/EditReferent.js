import ListErrors from '../ListErrors';
import React from 'react';
import agent from '../../agent';
import Request from 'superagent';
import { connect } from 'react-redux';
import Select from 'react-select';
import {
  SETTINGS_PAGE_UNLOADED,
  REFERENT_UPDATED,
  ASSOCIATE_REFERENT,
  ASSOCIATE_REFERENT_PREFERENCE,
  ASSOCIATE_REFERENT_CLEAR_PREFERENCE
} from '../../constants/actionTypes';

export class EditReferentForm extends React.Component {
  constructor() {
    super();

    this.state = {
      id: '',
      firstname: '',
      lastname: '',
      title: '',
      workPhone: '',
      cellPhone: '',
      fax: '',
      email: '',
      selectedOptions: null,
      selectedPrefs: null,
      preferencesArray: []
    };

    this.updateState = field => ev => {
      const state = this.state;
      const newState = Object.assign({}, state, { [field]: ev.target.value });
      this.setState(newState);
    };


    this.submitForm = ev => {
      ev.preventDefault();

      const referent = Object.assign({
        id: this.state.id,
        firstname: this.state.firstname,
        lastname: this.state.lastname,
        title: this.state.title,
        workPhone: this.state.workPhone,
        cellPhone: this.state.cellPhone,
        fax: this.state.fax,
        email: this.state.email
      });

      this.props.onSubmitForm(referent);

      this.props.onAssociate(this.state.organismesSelected, referent.email);

      this.props.onClearPreference(referent.email);

      this.props.onPreference(this.state.preferencesSelected, referent.email);
    };
  }

  handleChangePreferences = (selectedOptions) => {
    this.setState({ 'preferencesSelected': selectedOptions });
    console.log(`Preference selected:`, selectedOptions);
  };

  handleChangeOrganismes = (selectedOptions) => {
    this.setState({ 'organismesSelected': selectedOptions });
    console.log(`Option selected:`, selectedOptions);
  };

  componentWillMount() {
    if (this.props.currentReferent) {
      Object.assign(this.state, {
        id: this.props.currentReferent.id,
        firstname: this.props.currentReferent.firstname,
        lastname: this.props.currentReferent.lastname,
        title: this.props.currentReferent.title,
        workPhone: this.props.currentReferent.workphone,
        cellPhone: this.props.currentReferent.cellphone,
        fax: this.props.currentReferent.fax,
        email: this.props.currentReferent.email
      });
    }
    var url = "http://localhost:4000/organismereferents"
    Request.get(url).then((response) => {
      console.log(response.body);
      var data = response.body.rows;
      this.setState({ 'organismesReferent': data });
    });
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.currentReferent) {
      this.setState(Object.assign({}, this.state, {
        id: nextProps.currentReferent.id,
        firstname: nextProps.currentReferent.firstname,
        lastname: nextProps.currentReferent.lastname,
        title: nextProps.currentReferent.title,
        workPhone: nextProps.currentReferent.workphone,
        cellPhone: nextProps.currentReferent.cellphone,
        fax: nextProps.currentReferent.fax,
        email: nextProps.currentReferent.email
      }));
      var currentOrganismes = [];
      for (var i = 0; i < nextProps.currentOrganismeReferents.length; i++) {
        currentOrganismes.push({
          value: nextProps.currentOrganismeReferents[i].OrganismeId,
          label: nextProps.currentOrganismeReferents[i].OrganismeName
        });
      }
      var currentPreferences = [];
      for (var j = 0; j < nextProps.currentPreferences.length; j++) {
        currentPreferences.push({
          value: nextProps.currentPreferences[j].PreferenceId,
          label: nextProps.currentPreferences[j].PreferenceType
        });
      }
      this.setState({ 'organismesSelected': currentOrganismes });
      this.setState({ 'preferencesSelected': currentPreferences });

      var url = "http://localhost:4000/organismereferents"
      Request.get(url).then((response) => {
        console.log(response.body);
        var data = response.body.rows;
        this.setState({ 'organismesReferent': data });
      });

      var preferenceUrl = "http://localhost:4000/preferencesreferent"
      Request.get(preferenceUrl).then((response) => {
        console.log(response.body);
        var data = response.body.Preferences;
        this.setState({ 'preferencesArray': data });
      });
    }
  }

  render() {
    const currentReferent = this.props.currentReferent;
    if (!currentReferent) {
      return null;
    }

    const organismesReferent = [];
    if (this.state != null && this.state.organismesReferent != null) {
      Object.values(this.state.organismesReferent).map((d) =>
        organismesReferent.push({ value: d.Id, label: d.Name })
      );
    }
    const preferences = [];
    if (this.state != null && this.state.preferencesArray != null) {
      Object.values(this.state.preferencesArray).map((d) =>
        preferences.push({ value: d.Id, label: d.Type })
      );
    }
    return (
      <form onSubmit={this.submitForm}>
        <fieldset>

          <fieldset className="form-group">
            <input
              className="form-control form-control-lg"
              type="text"
              placeholder="Prénom"
              value={this.state.firstname}
              onChange={this.updateState('firstname')} />
          </fieldset>

          <fieldset className="form-group">
            <input
              className="form-control form-control-lg"
              type="text"
              placeholder="Nom de famille"
              value={this.state.lastname}
              onChange={this.updateState('lastname')} />
          </fieldset>

          <fieldset className="form-group">
            <input
              className="form-control form-control-lg"
              type="text"
              placeholder="Titre"
              value={this.state.title}
              onChange={this.updateState('title')} />
          </fieldset>

          <fieldset className="form-group">
            <input
              className="form-control form-control-lg"
              type="phone"
              placeholder="Tel. Travail"
              value={this.state.workPhone}
              onChange={this.updateState('workPhone')} />
          </fieldset>

          <fieldset className="form-group">
            <input
              className="form-control form-control-lg"
              type="phone"
              placeholder="Tel. Cellulaire"
              value={this.state.cellPhone}
              onChange={this.updateState('cellPhone')} />
          </fieldset>

          <fieldset className="form-group">
            <input
              className="form-control form-control-lg"
              type="phone"
              placeholder="Fax"
              value={this.state.fax}
              onChange={this.updateState('fax')} />
          </fieldset>

          <fieldset className="form-group">
            <input
              className="form-control form-control-lg"
              type="email"
              placeholder="Email"
              value={this.state.email}
              onChange={this.updateState('email')} />
          </fieldset>
          <br/>
          <fieldset className="form-group">
            <h3 className="text-xs-center">Association</h3>
            <Select
              placeholder="Organismes référent"
              className="form-control form-control-xl"
              value={this.state.organismesSelected}
              selectedOptions={this.state.organismesSelected}
              options={organismesReferent}
              isMulti={true}
              isSearchable={true}
              onChange={this.handleChangeOrganismes.bind(this)}>
              )}
                    </Select>
          </fieldset>
          <br />

          <fieldset className="form-group">
            <h3 className="text-xs-center">Préférence Communication</h3>
            <Select
              placeholder="Preferences"
              className="form-control form-control-xl"
              value={this.state.preferencesSelected}
              options={preferences}
              selectedOptions={this.state.preferencesSelected}
              isMulti={true}
              isSearchable={true}
              onChange={this.handleChangePreferences.bind(this)}>
              )}
                    </Select>
          </fieldset>

          <button
            className="btn btn-lg btn-primary btn-block"
            type="submit"
            disabled={this.state.inProgress}>
            Mettre à jour
          </button>

        </fieldset>
      </form>
    );
  }
}
const mapStateToProps = state => ({
    ...state.settings,
    currentReferent: state.common.currentReferent,
    organismesSelected: [],
    preferencesSelected: []
});

const mapDispatchToProps = dispatch => ({
    onSubmitForm: referent =>
        dispatch({ type: REFERENT_UPDATED, payload: agent.Referent.save(referent) }),
    onPreference: (preferencesSelected, email) => {
        preferencesSelected.forEach(function (element) {
            const payload = agent.Referent.preference(element.value, email);
            dispatch({ type: ASSOCIATE_REFERENT_PREFERENCE, payload })
        });
    },
    onClearPreference: (email) => {
        const payload = agent.Referent.clear(email);
        dispatch({ type: ASSOCIATE_REFERENT_CLEAR_PREFERENCE, payload})
    },
    onUnload: () => dispatch({ type: SETTINGS_PAGE_UNLOADED }),
    onAssociate: (organismesSelected, email) => {
        if (organismesSelected.length > 0) {
            organismesSelected.forEach(function (element) {
                const payload = agent.Referent.associate(element.value, email);
                dispatch({ type: ASSOCIATE_REFERENT, payload })
            });
        } else {
            const payload = agent.Referent.associate(0, email);
            dispatch({ type: ASSOCIATE_REFERENT, payload })
        }
    }
});

class EditReferent extends React.Component {
    constructor() {
        super();
        this.state = {
            currentReferent: '',
            organismes: [],
            preferenceTypes: []
        }
    }

    componentWillMount() {
        var associationUrl = "http://localhost:4000/referent/associations/" + this.props.match.params.email;
        Request.get(associationUrl).then((response) => {
            console.log(response.body);
            var data = response.body;
            this.setState({ 'organismes': data.organismeReferents });
            this.setState({ 'preferenceTypes': data.preferenceTypes });
        });
        var url = "http://localhost:4000/referent/" + this.props.match.params.email;
        Request.get(url).then((response) => {
            console.log(response.body);
            var data = response.body;
            this.setState({ 'currentReferent': data.referent });
        });
    }

    render() {
        return (
            <div className="settings-page">
                <div className="container page">

                    <hr />
                    <h1 className="text-xs-center">Paramètres du Référent</h1>
                    <hr />
                    <br />
                    <br />

                    <div className="row">
                        <div className="col-md-8 offset-md-2 col-xs-12">

                            <ListErrors errors={this.props.errors}></ListErrors>
                            <EditReferentForm
                                currentReferent={this.state.currentReferent}
                                currentOrganismeReferents={this.state.organismes}
                                currentPreferences={this.state.preferenceTypes}
                                onSubmitForm={this.props.onSubmitForm}
                                onPreference={this.props.onPreference}
                                onClearPreference={this.props.onClearPreference}
                                onAssociate={this.props.onAssociate} />
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditReferent);