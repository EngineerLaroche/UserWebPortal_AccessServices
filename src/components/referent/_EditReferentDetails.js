//***TEMPORAIRE***
//Utilisé pour éditer le Référent à partir de la liste des Référents
//Ce fichier est dans le fond une copie du fichier SettingsReferent.js

import ListErrors from './ListErrors';
import React from 'react';
import agent from '../../agent';
import { connect } from 'react-redux';
import {
  SETTINGS_SAVED,
  SETTINGS_PAGE_UNLOADED
} from '../constants/actionTypes';

class EditReferentForm extends React.Component {
  constructor() {
    super();

    this.state = {
      firstname: '',
      lastname: '',
      title: '',
      workPhone: '',
      cellPhone: '',
      fax: '',
      email: ''
    };

    this.updateState = field => ev => {
      const state = this.state;
      const newState = Object.assign({}, state, { [field]: ev.target.value });
      this.setState(newState);
    };

    this.submitForm = ev => {
      ev.preventDefault();

      const referent = Object.assign({}, this.state);
      this.props.onSubmitForm(referent);
    };
  }

  componentWillMount() {
    if (this.props.currentReferent) {
      Object.assign(this.state, {
        firstname: this.props.currentReferent.firstname,
        lastname: this.props.currentReferent.lastname,
        title: this.props.currentReferent.title,
        workPhone: this.props.currentReferent.workPhone,
        cellPhone: this.props.currentReferent.cellPhone,
        fax: this.props.currentReferent.fax,
        email: this.props.currentReferent.email
      });
    }
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.currentReferent) {
      this.setState(Object.assign({}, this.state, {

        firstname: nextProps.currentReferent.firstname,
        lastname: nextProps.currentReferent.lastname,
        title: nextProps.currentReferent.title,
        workPhone: nextProps.currentReferent.workPhone,
        cellPhone: nextProps.currentReferent.cellPhone,
        fax: nextProps.currentReferent.fax,
        email: nextProps.currentReferent.email
      }));
    }
  }

  render() {
    return (
      <form onSubmit={this.submitForm}>
        <fieldset>

          <fieldset className="form-group">
            <input
              className="form-control form-control-lg"
              type="text"
              placeholder="Prénom"
              value={this.state.firstname}
              onChange={this.updateState('firstname')} />
          </fieldset>

          <fieldset className="form-group">
            <input
              className="form-control form-control-lg"
              type="text"
              placeholder="Nom de famille"
              value={this.state.lastname}
              onChange={this.updateState('lastname')} />
          </fieldset>

          <fieldset className="form-group">
            <input
              className="form-control form-control-lg"
              type="text"
              placeholder="Titre"
              value={this.state.title}
              onChange={this.updateState('title')} />
          </fieldset>

          <fieldset className="form-group">
            <input
              className="form-control form-control-lg"
              type="phone"
              placeholder="Tel. Travail"
              value={this.state.workPhone}
              onChange={this.updateState('workPhone')} />
          </fieldset>

          <fieldset className="form-group">
            <input
              className="form-control form-control-lg"
              type="phone"
              placeholder="Tel. Cellulaire"
              value={this.state.cellPhone}
              onChange={this.updateState('cellPhone')} />
          </fieldset>

          <fieldset className="form-group">
            <input
              className="form-control form-control-lg"
              type="phone"
              placeholder="Fax"
              value={this.state.fax}
              onChange={this.updateState('fax')} />
          </fieldset>

          <fieldset className="form-group">
            <input
              className="form-control form-control-lg"
              type="email"
              placeholder="Email"
              value={this.state.email}
              onChange={this.updateState('email')} />
          </fieldset>

          <button
            className="btn btn-lg btn-primary btn-block"
            type="submit"
            disabled={this.state.inProgress}>
            Mettre à jour
          </button>

        </fieldset>
      </form>
    );
  }
}

const mapStateToProps = state => ({
  ...state.settings,
  currenReferent: state.common.currentReferent
});

//A retirer: agent.Auth  ?
const mapDispatchToProps = dispatch => ({
  onSubmitForm: referent =>
    dispatch({ type: SETTINGS_SAVED, payload: agent.Auth.save(referent) }),
  onUnload: () => dispatch({ type: SETTINGS_PAGE_UNLOADED })
});

class EditReferentDetails extends React.Component {
  render() {
    return (
      <div className="settings-page">
        <div className="container page">

              <hr/>
              <h1 className="text-xs-center">Paramètres du Référent</h1>
              <hr/>
              <br/>
              <br/>

          <div className="row">
            <div className="col-md-8 offset-md-3 col-xs-12">

              <ListErrors errors={this.props.errors}></ListErrors>
              <EditReferentForm
                currentReferent={this.props.currentRefrent}
                onSubmitForm={this.props.onSubmitForm} />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

//connect ?
export default connect(mapStateToProps, mapDispatchToProps)(EditReferentDetails);
