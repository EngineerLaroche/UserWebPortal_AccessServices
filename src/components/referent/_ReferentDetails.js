//***TEMPORAIRE***
//Utilisé pour accéder et afficher les info du Référent à partir de la liste des Référents
//Ce fichier est dans le fond presqu'une copie du fichier ProfileReferent.js

import React from 'react';
import { Link } from 'react-router-dom';
import Request from 'superagent';
import { connect } from 'react-redux';
import agent from '../../agent';
import {
  DEACTIVATE,
  REACTIVATE
} from '../../constants/actionTypes';

const EditReferentSettings = props => {
  if (props.isReferent) {
    return (
      <div>
        <Link to={`/referent/edit/${props.referent.email}`} className="btn btn-sm btn-outline-secondary action-btn">
          <i className="ion-gear-a">
          </i>
          Modifier Référent
          </Link>
        {(props.hasHighRights && !props.isActivated) &&
          <button onClick={props.onClickReactivate} className="btn btn-sm btn-outline-secondary action-btn">
            <i className="ion-gear-a">
            </i>
            Réactiver Référent
          </button>
        }
        {(props.hasHighRights && props.isActivated) &&
          <button onClick={props.onClickDeactivate} className="btn btn-sm btn-outline-secondary action-btn">
            <i className="ion-gear-a">
            </i>
            Désactiver Référent
          </button>
        }
      </div>
    );
  }
  return null;
};

const mapStateToProps = state => ({
  currentReferent: state.common.currentReferent,
  referent: state.referent
});

const mapDispatchToProps = dispatch => ({

  //agent.Auth ??
  onClickDeactivate: (email) => dispatch({ type: DEACTIVATE, payload: agent.Auth.deactivate(email) }),
  onClickReactivate: (email) => dispatch({ type: REACTIVATE, payload: agent.Auth.reactivate(email) })
});

class ReferentDetails extends React.Component {

  componentWillMount() {
    var url = "http://localhost:4000/referent/" + this.props.match.params.email;
    Request.get(url).then((response) => {
      console.log(response.body);
      var data = response.body.referent;
      this.setState(data);
    });
  }

  render() {
    const referent = this.state;
    if (!referent) {
      return null;
    }

    //Verifier cette constante...
    const isReferent = this.props.currentReferent;
    
    const isActivated = (referent.active === 0);
    const name = referent.firstname + " " + referent.lastname;


    // A utiliser pour la selection (comment contacter referent) entre workPhone, cellPhone et Fax....
    //
    // var role;
    // switch (profile.role) {
    //   case 1:
    //     role = "Directeur";
    //     break;
    //   case 2:
    //     role = "Coordonateur";
    //     break;
    //   case 3:
    //     role = "Adjoint-Coordonateur";
    //     break;
    //   default:
    //     role = "Intervenant";
    //     break;
    // }


    return (
      <div className="referent-page">
        <div className="referent-info">
        <br/>
          <br/>
          <h1 className="text-xs-center">Un Référent</h1>
          <br/>
          <br/>
          <div className="container">
            <div className="row">
              <div className="text-xs-left">

                <h4>Nom: {name}</h4>
                <h4>Email: {referent.email}</h4>
                <h4>Titre: {referent.titre}</h4>
                <br/>
                <h4>Telephone travail: {referent.workPhone}</h4>
                <h4>Telephone cellulaire: {referent.cellPhone}</h4>
                <h4>Fax: {referent.fax}</h4>
                <EditReferentSettings isReferent={true} 
                                      referent={referent} 
                                      isActivated={isActivated} 
                                      onClickDeactivate={ () => {this.props.onClickDeactivate(referent.email)}} 
                                      onClickReactivate={() => {this.props.onClickReactivate(referent.email)}}/>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(ReferentDetails);
export {ReferentDetails, mapStateToProps };
