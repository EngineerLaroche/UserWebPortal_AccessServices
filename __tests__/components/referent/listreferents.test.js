import React from 'react';
import {mount, shallow} from 'enzyme';
import {ListReferents} from 'components/referent/ListReferents';
import agent from 'agent';

describe('ListReferents is rendering', () => {
    let component;
    let data = {
        0: {
            CellPhone: "1-820-888-1111",
            Email: "referent@test.com",
            Fax: "123456789",
            Firstname: "Referent",
            Id: 1,
            Lastname: "Test",
            Title: "Avocat",
            WorkPhone: "454729692231"
        }
    };
    let handleChange = jest.fn();
    let searchReferents = jest.fn();

    beforeEach(() => {
        component = shallow(<ListReferents handleChange={handleChange}
                                           searchReferents={searchReferents}/>);
    });
    //
    it('Is mounting', () => {
        expect(component.setState({'referents': data}));
        expect(component.setState({'search': ''}));
        expect(component.find('input').props().value).toEqual(component.props().search);
        expect(component.find('li').length).toEqual(1);
        expect(component.find('li').props().key).toEqual(data.Email);
        expect(component).toMatchSnapshot();
    });
});

describe('ListReferent behaves', () => {
    let component;
    let data = {
        0: {
            CellPhone: "1-820-888-1111",
            Email: "referent@test.com",
            Fax: "123456789",
            Firstname: "Referent",
            Id: 1,
            Lastname: "Test",
            Title: "Avocat",
            WorkPhone: "454729692231"
        }
    };
    let handleChange = jest.fn();
    let searchReferents = jest.fn();

    beforeEach(() => {
        component = shallow(<ListReferents handleChange={handleChange}
                                           searchReferents={searchReferents}/>);
    });

    // On search
    it('OnSearch should be called properly', () => {
        const button = component.find('button').first();
        const searchInput = component.find('input').first();

        searchInput.simulate('change', {target: {value: 'ref'}, param : 'search'});
        expect(component.setState({[search] : 'ref'})).toHaveBeenCalledTimes(1);
        button.simulate('click');
        expect(searchReferents).toHaveBeenCalledTimes(1);
    });

    // On click on list item
    it('Click the list item links, should redirect rightfully', () => {
        component.setState({'referents': data});
        Object.values(data).map((d, i) => {
            let link = component.find('li').at(i).children().first();
            console.log(link.debug());
            expect(link.props().to)
                .toEqual('/referent/' + d.Email);
        });
    });
});