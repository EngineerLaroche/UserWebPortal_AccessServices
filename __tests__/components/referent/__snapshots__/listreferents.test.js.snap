// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListReferents is rendering Is mounting 1`] = `
<div
  className="container page"
>
  <hr />
  <h1
    className="text-xs-center"
  >
    Liste des Référents
  </h1>
  <hr />
  <br />
  <br />
  <div
    className="row"
  >
    <div
      className="col-md-8 offset-md-3 col-xs-12"
    >
      <div
        className="text-xs-center"
      >
        <label>
          Search: 
        </label>
        <br />
        <fieldset>
          <fieldset
            className="form-group"
          >
            <input
              onChange={[Function]}
              type="text"
            />
            <button
              onClick={[Function]}
            >
              Go
            </button>
          </fieldset>
        </fieldset>
      </div>
      <br />
      <br />
      <div
        className="text-xs-left"
      >
        <li
          key="referent@test.com"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/referent/referent@test.com"
          >
            Referent
             
            Test
          </Link>
          <hr />
        </li>
      </div>
    </div>
  </div>
</div>
`;
